src/main/java/com/palantir/gradle/gitversion/JGitDescribe.java

```diff
@@ -88,16 +91,19 @@ public String describe(String prefix) {
         // Maps commit hash to list of all refs pointing to given commit hash.
         // All keys in this map should be same as commit hashes in 'git show-ref --tags -d'
         Map<String, RefWithTagName> commitHashToTag = new HashMap<>();
-        for (Map.Entry<String, Ref> entry : git.getRepository().getTags().entrySet()) {
-            RefWithTagName refWithTagName = new RefWithTagName(entry.getValue(), entry.getKey());
-
-            ObjectId peeledRef = refWithTagName.getRef().getPeeledObjectId();
-            if (peeledRef == null) {
-                // lightweight tag (commit object)
-                updateCommitHashMap(commitHashToTag, comparator, entry.getValue().getObjectId(), refWithTagName);
+        Repository repository = git.getRepository();
+        for (Map.Entry<String, Ref> entry : repository.getTags().entrySet()) {
+            Ref peeledRef = repository.peel(entry.getValue());
```

Comment:
Found a bug while writing unit tests. This is done for us when we run `git.describe().call()`. But we shouldn't depend on that. Peeling ref means following links and actually getting peeled object id for this ref. Otherwise, `peeledRef.getPeeledObjectId()` would just return null

Reply:
😱